source/agora/common/Serializer.d

```diff
@@ -119,21 +123,24 @@ public void serializePart (ubyte record, scope SerializeDg dg)
 public void serializePart (ushort record, scope SerializeDg dg)
     @trusted
 {
-    dg((cast(ubyte*)&record)[0 .. ushort.sizeof]);
+    dg(toVarInt(record));
 }
 
 /// Ditto
 public void serializePart (uint record, scope SerializeDg dg)
     @trusted
 {
-    dg((cast(ubyte*)&record)[0 .. uint.sizeof]);
+    dg(toVarInt(record));
 }
 
 /// Ditto
-public void serializePart (ulong record, scope SerializeDg dg)
+public void serializePart (ulong record, scope SerializeDg dg, CompactMode compact = CompactMode.Yes)
```

Comment:
For this `/// Ditto` to work properly, you will need to add the documentation of `compact` into the `serializeFull` method.

Reply:
Added documentation.